src/espIdf/serial/serialPort.ts

```diff
@@ -87,24 +86,50 @@ export class SerialPort {
   private list(workspaceFolder: vscode.Uri): Thenable<SerialPortDetails[]> {
     return new Promise(async (resolve, reject) => {
       try {
+        const listOfSerialPorts = await SerialPortLib.SerialPort.list();
+
+        const choices = listOfSerialPorts.map((item) => {
+          return new SerialPortDetails(
+            item.path,
+            item.manufacturer,
+            item.vendorId,
+            item.productId
+          );
+        });
         const pythonBinPath = idfConf.readParameter(
           "idf.pythonBinPath",
           workspaceFolder
         ) as string;
-        const buff = await spawn(pythonBinPath, ["get_serial_list.py"]);
-        const regexp = /\'(.*?)\'/g;
-        const arrayPrint = buff.toString().match(regexp);
-        const choices: SerialPortDetails[] = Array<SerialPortDetails>();
+        const idfPath = idfConf.readParameter(
+          "idf.espIdfPath",
+          workspaceFolder
+        );
+        const esptoolPath = join(
+          idfPath,
+          "components",
+          "esptool_py",
+          "esptool",
+          "esptool.py"
+        );
+        async function processPorts(serialPort: SerialPortDetails) {
+          try {
+            const chipIdBuffer = await spawn(
+              pythonBinPath,
+              [esptoolPath, "--port", serialPort.comName, "chip_id"],
+              {},
+              2000 // success is quick, failing takes too much time
+            );
+            const regexp = /Chip is(.*?)[\r]?\n/;
+            const chipIdString2 = chipIdBuffer.toString().match(regexp);
 
-        if (arrayPrint) {
-          arrayPrint.forEach((portStr) => {
-            const portChoice = portStr.replace(/'/g, "").trim();
-            choices.push(new SerialPortDetails(portChoice));
-          });
-          resolve(choices);
-        } else {
-          reject(new Error("No serial ports found"));
+            serialPort.chipType = chipIdString2[1].trim();
```

Comment:
Could we add validation that match exists and it has at least 2 elements here ?

Reply:
good idea!